### PR TITLE
feat: add support to specify targets or remove targets when running mitosis build

### PR DESCRIPTION
**Pull request by findlay-best-wishes**
## Description
feature for [#509](https://github.com/BuilderIO/mitosis/issues/509)
- Adding support to specify targets or remove targets when running mitosis build, as the following usage.
- Adding docs for mitosis-build-command

``` javascript
// mitosis.config.js
module.exports = {
  targets: ['react', 'vue']
}
```

``` sh
# sh
mitosis build --targets react,svelte  # => ['react', 'vue', 'svelte']

mitosis build --targets-ignored react,svelte  # => ['vue']

mitosis build --targets svelte --targets-ignored  react  # => ['vue', 'svelte']

mitosis build --targets svelte --targets-ignored  svelte  # => ['react', 'vue']

mitosis build  ----targets-ignored  svelte --targets svelte  # => ['react', 'vue']

mitosis build  --targets react --targets-ignored  react  # => ['vue']

mitosis build  --targets-ignored  react --targets react  # => ['vue']

```

the priority of targets option: 
1. *option-ignoreTargets of cli*  (highest priority）
2. *option-targets of cli*  (medium priority)
3. *option-targets of config file*  (minimum priority)

### CODE REVIEW

File: packages/cli/src/commands/build.ts [hunk 1]
```diff
@@ -1,15 +1,36 @@
-import { GluegunCommand } from 'gluegun';
+import { GluegunCommand, GluegunParameters } from 'gluegun';
+import { MitosisConfig, Target } from '@builder.io/mitosis';
 import { build } from '../build/build';
 import { getMitosisConfig } from '../helpers/get-mitosis-config';
 
+const getTargets = (mitosisConfig: MitosisConfig, cliOpts: GluegunParameters['options']) => {
+  const targetsFromCli: Target[] = (cliOpts.targets || '').split(',');
+  const targetsIgnoredMap: Record<Target, true> = (cliOpts.targetsIgnored || '')
```
Comment (samijaber) [inline packages/cli/src/commands/build.ts:8]: I would rather call this option `excludeTargets` (active verb tense)

**Review by samijaber — COMMENTED** [review]
Nice! Thanks for adding this.

File: packages/cli/src/commands/build.ts (hunk 1 above)
Comment (findlay-best-wishes) [reply packages/cli/src/commands/build.ts:8]: > I would rather call this option `excludeTargets` (active verb tense)

okay

File: docs/cli.md [hunk 2]
```diff
@@ -10,15 +10,25 @@ We currently have two CLI commands: `mitosis build` and `mitosis compile`.
 - Receives 1 Mitosis component file as input
 - Outputs it to 1 designated target.
 
+You can get more information by `mitosis --help`
```
Comment (samijaber) [inline docs/cli.md:13]:
```suggestion
You can get more information by running `mitosis --help`
```

File: docs/cli.md [hunk 3]
```diff
@@ -10,15 +10,25 @@ We currently have two CLI commands: `mitosis build` and `mitosis compile`.
...
   - transpiles them as-is to JS
 - Performs necessary transformations to both Mitosis & non-Mitosis files so that the output folder is coherent and valid (like renaming all component imports in a Svelte target such that they match the output name, ending in `.svelte`)
+
+### options
+
+|                             Option                              | Description                                           | Example                                                                                          |
+| :-------------------------------------------------------------: | :---------------------------------------------------- | :----------------------------------------------------------------------------------------------- |
+|       <p style="white-space:nowrap">--config=\<file\></p>       | To specify config file, defaults to mitosis.config.js | none                                                                                             |
+|     <p style="white-space:nowrap">--targets=[format...]</p>     | To specify extra build targets                        | `mitosis build --targets react,vue,svelte` will add 'react','vue' and 'svelte' to build targets. |
```
Comment (samijaber) [inline docs/cli.md:33]:
```suggestion
|     <p style="white-space:nowrap">--targets=[format...]</p>     | To specify extra build targets                        | `mitosis build --targets react,vue,svelte` will add 'react', 'vue' and 'svelte' to build targets. |
```

**Comment by samijaber** [comment]
Thank you for adding this!

**Pull request merged**